data/src/main/java/io/gnosis/kouban/data/managers/SearchManager.kt

```diff
@@ -78,8 +79,9 @@ data class TransactionTimestampFilter(
 ) : BoundaryFilter<Transaction>() {
 
     override fun apply(item: Transaction): Boolean {
-        return with(item.timestampAsDate()) {
-            (upperBound == null || before(upperBound)) && (lowerBound == null || after(lowerBound))
+        return with(item.timestampAsDate().onlyDate()) {
+            (upperBound == null || upperBound?.onlyDate() == this || before(upperBound))
+                    && (lowerBound == null || lowerBound?.onlyDate() == this || after(lowerBound))
```

Comment:
this check should be encapsulated in a subroutine. Date?.isBeforeOrEqual(Date?), Date?.isAfterOrEqual(Date?)